crates/typst/src/geom/gradient.rs

```diff
@@ -0,0 +1,1089 @@
+///
+/// ## Relativeness
+///
+/// Gradients can be relative to either the shape they are painted on, or to the
+/// nearest parent containers. This is controlled by the `relative` argument of the
+/// constructors. By default, gradients are relative to the shape they are painted on,
+/// unless they are painted on text, in which case they are relative to the parent.
+///
+/// The way the parent is determined is as follows:
+/// - For shapes that are placed at the root/top level of the document, the parent
+///   is the page itself.
+/// - For other shapes, the parent is the innermost [`block`]($block) or [`box`]($box)
+///   that contains the shape. This includes the boxes and blocks that are implicitly
+///   created by show rules. For example, a [`rotate`]($rotate) will not affect the
+///   parent of a gradient, but a [`grid`]($grid) will.
+///
+/// ## Color spaces and interpolation
+///
+/// Gradients can be interpolated in any color space. By default, gradients are
+/// interpolated in the [Oklab]($color.oklab) color space, which is a perceptually
+/// uniform color space. This means that the gradient will be perceived as having
+/// a uniform progression of colors. This is particularly useful for data
+/// visualization.
+///
+/// However, you can choose to interpolate the gradient in any supported color
+/// space you want, but beware that some color spaces are not suitable for
+/// perceptually interpolating between colors. Below you can find a list of
+/// color spaces and whether they are suitable for perceptual interpolation.
+///
+/// |           Color space           | Perceptually uniform? |
+/// | ------------------------------- | --------------------- |
+/// |      [Oklab]($color.oklab)      |           ✅          |
+/// |      [sRGB]($color.rgb)         |           ❌          |
+/// | [linear-RGB]($color.linear-rgb) |           ✅          |
+/// |      [CMYK]($color.cmyk)        |           ❌          |
+/// |     [Grayscale]($color.luma)    |           ✅          |
+/// |       [HSL]($color.hsl)         |           ❌          |
+/// |       [HSV]($color.hsv)         |           ❌          |
+///
+/// ```example
+/// #set text(fill: white)
+/// #set block(spacing: 0pt)
+///
+/// #let spaces = (
+///   ("Oklab", color.oklab),
+///   ("sRGB", color.rgb),
+///   ("linear-RGB", color.linear-rgb),
+///   ("CMYK", color.cmyk),
+///   ("Grayscale", color.luma),
+///   ("HSL", color.hsl),
+///   ("HSV", color.hsv),
+/// )
+///
+/// #for space in spaces {
+///   block(
+///     width: 100%,
+///     height: 10pt,
+///     fill: gradient.linear(red, blue, space: space.at(1))
+///   )[
+///     #space.at(0)
+///   ]
+/// }
+/// ```
+///
+/// ## Direction
+///
+/// Some gradients are sensitive to the direction of the gradient. For example, a
+/// linear gradient has an angle that determines the direction of the gradient. Instead
+/// of the traditional clockwise angle, Typst uses an anti-clockwise angle, with 0° being
```

Comment:
why do we go against the tradition?

Reply:
Because we're not and that snippet of the docs is wrong?